---
bool success
Action last_action
Condition[] expected_state
